commit 7ec3e95e7a778f1202f85a9046917d4f231ce799
Author: Gustavo A. R. Silva <gustavoars@kernel.org>
Date:   Wed Jul 22 13:43:58 2020 -0500

    tg3: Avoid the use of one-element array
    
    One-element arrays are being deprecated[1]. Replace the one-element
    array with a simple value type 'u32 reserved2'[2], once it seems
    this is just a placeholder for alignment.
    
    [1] https://github.com/KSPP/linux/issues/79
    [2] https://github.com/KSPP/linux/issues/86
    
    Tested-by: kernel test robot <lkp@intel.com>
    Link: https://github.com/GustavoARSilva/linux-hardening/blob/master/cii/0-day/tg3-20200718.md
    Signed-off-by: Gustavo A. R. Silva <gustavoars@kernel.org>
    Reviewed-by: Michael Chan <michael.chan@broadcom.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/ethernet/broadcom/tg3.h b/drivers/net/ethernet/broadcom/tg3.h
index 6953d0546acb..1000c894064f 100644
--- a/drivers/net/ethernet/broadcom/tg3.h
+++ b/drivers/net/ethernet/broadcom/tg3.h
@@ -2847,7 +2847,7 @@ struct tg3_ocir {
 	u32				port1_flags;
 	u32				port2_flags;
 	u32				port3_flags;
-	u32				reserved2[1];
+	u32				reserved2;
 };